test/functional/unattended_controller_test.rb

```diff
@@ -358,12 +358,47 @@ class UnattendedControllerTest < ActionController::TestCase
   test "should return and log error when template not found" do
     @request.env["REMOTE_ADDR"] = @ub_host.ip
     Host::Managed.any_instance.expects(:provisioning_template).returns(nil)
-    Rails.logger.expects(:error).with(regexp_matches(/unable to find provision template/))
+    Rails.logger.expects(:error).with(regexp_matches(/unable to find.*template/))
     get :host_template, {:kind => 'provision'}
     assert_response :not_found
   end
 
   test 'host template provision URL can be generated from routes' do
-    assert_routing '/unattended/provision', {:controller => 'unattended', :action => 'host_template', :kind => 'provision'}
+    assert_routing '/unattended/provision', {:controller => 'unattended', :action => 'host_template', :kind => 'provision', :format => 'text'}
+  end
+
+  test "should not render a template to anonymous user" do
+    get :host_template, {:kind => 'PXELinux', :spoof => @rh_host.ip, :format => 'text'}
+    assert_response :redirect
+  end
+
+  test "should not render a template to user w/o email" do
+    user = FactoryGirl.create(:user)
+    get :host_template, {:kind => 'PXELinux', :spoof => @rh_host.ip, :format => 'text'}, set_session_user(user)
+    assert_response :unprocessable_entity
+  end
+
+  test "should not render a template to unauthorized user" do
+    user = FactoryGirl.create(:user, :with_mail)
+    user.auth_source.stubs(:authenticated?).returns(false)
```

Comment:
I don't think this is needed or should be here, it looks like the test is meant to authenticate the user, but they shouldn't be authorised - not that authentication is failing.